Lesson5/Sulima Denis Homework 3/src/functions.cpp

```diff
@@ -0,0 +1,70 @@
+#include "functions.h"
+#include <set>
+#include <map>
+#include <algorithm>
+#include <sstream>
+
+struct cmp
+{
+    bool operator() (std::string lhs, std::string rhs)
+    {
+        std::string tmp_lhs {lhs};
+        std::string tmp_rhs {rhs};
+        std :: transform( tmp_lhs.begin(), tmp_lhs.end(), tmp_lhs.begin(), ::tolower );
+        std :: transform( tmp_rhs.begin(), tmp_rhs.end(), tmp_rhs.begin(), ::tolower );
+        if(tmp_lhs < tmp_rhs)
```

Comment:
_return tmp_lhs < tmp_rhs_

Reply:
std::transform is transformation tmp_lhs and tmp_rhs to lower register,  if return tmp_lhs < tmp_rhs that save it on lower register. So need to return the unchanged string

Reply:
I'm talking about "return true" and "return false"

Reply:
This is logical, I don’t know why I wrote this